### PR TITLE
FEATURE_Front_Add_Recipe_Ingredient

### PR DESCRIPTION
**Pull request by DoniaMroua**
In order to communicate with data access layer (Backend), We added coffeeScript factory called "datacontext" on which we grouped all CRUD functions of all entities.

### CODE REVIEW

**Comment by DoniaMroua** [comment]
@achr3f  @SamiDghim  @melas123  @medBelaid 
Merci de vérifier le code pour que je puisse faire le merge.

File: app/controllers/ingredient_recipe_controller.rb [hunk 1]
```diff
@@ -4,7 +4,9 @@ def create
     IngredientRecipe.create(recipe_id: params[:recipe_id],
                                    ingredient_id: params[:ingredient_id],
                                    quantity: params[:quantity],
-                                   unit: params[:unit])
+                                   mass_unit: params[:mass_unit],
```
Comment (SamiDghim) [inline app/controllers/ingredient_recipe_controller.rb]: plz don't forget spaces

File: .byebug_history [hunk 2]
```diff
@@ -0,0 +1,4 @@
+continue
```
Comment (SamiDghim) [inline .byebug_history:1]: add this file to .gitignore

File: app/assets/javascripts/controllers/auth/authCtrl.coffee [hunk 3]
```diff
@@ -1,8 +1,10 @@
-@cooking.controller 'authCtrl', ($scope, Auth, $location) ->
+@cooking.controller 'authCtrl', ($scope, $rootScope, Auth, $location) ->
   $scope.login = ->
     Auth.login( $scope.user ).then ->
       $location.path 'home'
+      $rootScope.auth = Auth.isAuthenticated()
```
Comment (achr3f) [inline app/assets/javascripts/controllers/auth/authCtrl.coffee:5]: first !  why using `$rootScope`,   second this variable is not used at all

File: app/assets/javascripts/controllers/home/home.coffee [hunk 4]
```diff
@@ -1,15 +1,199 @@
...
+  
+  #Refrech list after adding new recipe
+  $rootScope.$on 'ReloadRecipe', (event, data) ->
+    datacontext.getRecipes().success (data) ->
+      $rootScope.recipes = data.recipes
+ 
+  ###***************Modal Recipe*********************###
+  
+  $scope.OpenModalRecipe = (action,entity) ->
+    $rootScope.action = action
```
Comment (achr3f) [inline app/assets/javascripts/controllers/home/home.coffee:19]: Dead Code ??

File: app/assets/javascripts/controllers/nav/navCtrl.coffee [hunk 5]
```diff
@@ -15,3 +15,4 @@
 
   $scope.$on 'devise:logout', (e, user) ->
     $scope.user = {}
+    $rootScope.auth = Auth.isAuthenticated()
```
Comment (achr3f) [inline app/assets/javascripts/controllers/nav/navCtrl.coffee:18]: Dead code ?

File: app/models/ingredient_recipe.rb [hunk 6]
```diff
@@ -2,4 +2,7 @@ class IngredientRecipe < ActiveRecord::Base
   belongs_to :ingredient
   belongs_to :recipe
   validates :recipe_id,:ingredient_id,:quantity,:unit, presence: true
```
Comment (SamiDghim) [inline app/models/ingredient_recipe.rb:4]: remove the unit attribute if you will not use it.

**Pull request closed without merging**